### PR TITLE
Fix findTransactionsInPool in TestKit

### PR DESCRIPTION
**Pull request by MakarovS**
## Overview
Fix findTransactionsInPool in TestKi and add a test checking that node.submitTransaction adds transactions to the pool.

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java [hunk 1]
```diff
@@ -345,6 +348,31 @@ void createBlockWithTransactions() {
...
+      try {
+        service.getNode().submitTransaction(rawTransaction);
+        service.getNode().submitTransaction(rawTransaction2);
+      } catch (InternalServerError e) {
+        fail(e);
+      }
+
+      List<TransactionMessage> transactionsInPool =
+          testKit.findTransactionsInPool(
+              tx -> Arrays.equals(tx.getPayload(), messagePayload.getBytes(BODY_CHARSET)));
```
Comment (dmitry-timofeev) [inline exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java]: Would be shorter:

```suggestion
              tx -> Arrays.equals(tx.getPayload(), message.getPayload()));
```

Or maybe it will be simpler to use different ids?

File: exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java [hunk 2]
```diff
@@ -345,6 +348,31 @@ void createBlockWithTransactions() {
     }
   }
 
+  @Test
+  void findTransactionsInPool() {
```
Comment (dmitry-timofeev) [inline exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java]: Shall we also add a more focused test (because of #879) verifying that submitted through Node transactions appear in a pool (this one still needed to test findTransactionsInPool, but is more complex/won't provide as precise diagnostics in case native testkit or core breaks)?

File: exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java (hunk 1 above)
Comment (dmitry-timofeev) [inline exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java]: Also, if during migration of service ITs we will determine that this method is useful, we might add some `TransactionPreconditions`: `hasSamePayloadAs(TransactionMessage)`, `hasSamePayloadAs(RawTransaction)`, `hasServiceId(int)`, etc.

However, we must also consider what is better (e.g., in terms of code size, or error reporting) — let users use 
```
assertThat(testkit.getPool()).
    <Some library matcher, possibly, with our predicates>
```
or `findTransactionsInPool` :thinking:

Comment (MakarovS) [reply exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java]: Yes, this method proved useful, although almost always I wanted to just get all in-pool transactions of a certain service. But users might have other cases - so the current method seems good as it is for now.

File: exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java (hunk 2 above)
Comment (MakarovS) [reply exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java]: Added a test for that

File: exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java [hunk 3]
```diff
@@ -361,12 +407,14 @@ void createBlockWithTransactionsVarargs() {
...
-  private TransactionMessage constructTestTransactionMessage(String payload) {
+  private TransactionMessage constructTestTransactionMessage(String payload, TestKit testKit) {
+    EmulatedNode emulatedNode = testKit.getEmulatedNode();
+    KeyPair emulatedNodeKeyPair = emulatedNode.getServiceKeyPair();
     return TransactionMessage.builder()
         .serviceId(TestService.SERVICE_ID)
         .transactionId(TestTransaction.ID)
         .payload(payload.getBytes(BODY_CHARSET))
-        .sign(KEY_PAIR, CRYPTO_FUNCTION);
+        .sign(emulatedNodeKeyPair, CRYPTO_FUNCTION);
```
Comment (dmitry-timofeev) [inline exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java]: I think it is not appropriate to abuse this method in tests that do not _require_ transactions submitted by the service. Such transactions can be signed by any key, and passing testkit and extracting the key of the emulated node is irrelevant to these tests.

I'd make a couple of overloads:
```
constructTestTransactionMessage(payload)
constructTestTransactionMessage(payload, keypair|emulatedNode)
```

File: exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java [hunk 4]
```diff
@@ -387,11 +435,13 @@ private void checkTransactionsCommittedSuccessfully(
   void createBlockWithTransactionWithWrongServiceId() {
     try (TestKit testKit = TestKit.forService(TestServiceModule.class)) {
       short wrongServiceId = (short) (TestService.SERVICE_ID + 1);
+      EmulatedNode emulatedNode = testKit.getEmulatedNode();
+      KeyPair emulatedNodeKeyPair = emulatedNode.getServiceKeyPair();
```
Comment (dmitry-timofeev) [inline exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java]: (here and elsewhere) Why these changes? Doesn't `createBlockWithTransactions` accept any transactions?

File: exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java [hunk 5]
```diff
@@ -345,11 +347,55 @@ void createBlockWithTransactions() {
...
+    try (TestKit testKit = TestKit.forService(TestServiceModule.class)) {
+      TestService service = testKit.getService(TestService.SERVICE_ID, TestService.class);
+
+      TransactionMessage message = constructTestTransactionMessage("Test message", testKit);
+      RawTransaction rawTransaction = TestKit.toRawTransaction(message);
+
+      try {
+        service.getNode().submitTransaction(rawTransaction);
+      } catch (InternalServerError e) {
+        fail(e);
```
Comment (dmitry-timofeev) [inline exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java]: Just add `throws Exception` — JUnit will mark this test as an error in case it throws an exception.

File: exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java [hunk 6]
```diff
@@ -345,11 +347,55 @@ void createBlockWithTransactions() {
...
+      TransactionMessage message = constructTestTransactionMessage("Test message", testKit);
+      RawTransaction rawTransaction = TestKit.toRawTransaction(message);
+      TransactionMessage message2 = constructTestTransactionMessage("Test message 2", testKit);
+      RawTransaction rawTransaction2 = TestKit.toRawTransaction(message2);
+
+      try {
+        service.getNode().submitTransaction(rawTransaction);
+        service.getNode().submitTransaction(rawTransaction2);
+      } catch (InternalServerError e) {
+        fail(e);
```
Comment (dmitry-timofeev) [inline exonum-java-binding/testkit/src/test/java/com/exonum/binding/testkit/TestKitTest.java]: Same as ^
